### PR TITLE
upgrade: Add error message when staged deployment is incompatible

### PR DESCRIPTION
**Pull request by ckyrouac**
When running `rpm-ostree install <package>` without a reboot, `bootc upgrade` will fail. Prior to this, the error message was unclear of the cause.

fixes RHEL-36003

### CODE REVIEW

File: lib/src/cli.rs [hunk 1]
```diff
@@ -384,17 +384,31 @@ async fn upgrade(opts: UpgradeOpts) -> Result<()> {
...
-        return Err(anyhow::anyhow!(
-            "Booted deployment contains local rpm-ostree modifications; cannot upgrade via bootc. You can run `rpm-ostree reset` to undo the modifications."
-        ));
+        {
+            return Err(anyhow::anyhow!(
+                "Booted deployment contains local rpm-ostree modifications; cannot upgrade via bootc. You can run `rpm-ostree reset` to undo the modifications."
+            ));
+        }
+
+        if host.status.staged.is_some()
```
Comment (cgwalters) [inline lib/src/cli.rs:399]: We should be able to drop this part; I know Rust combinators can be quite obscure but it's basically what the `map_or(false)` is doing here...if the option is None, then the result will be `false`.

File: lib/src/cli.rs [hunk 2]
```diff
@@ -384,17 +384,31 @@ async fn upgrade(opts: UpgradeOpts) -> Result<()> {
...
+            && host
+                .status
+                .staged
+                .as_ref()
+                .map_or(false, |b| b.incompatible)
```
Comment (cgwalters) [inline lib/src/cli.rs:404]: But for a bigger cleanup, I think we can merge the logic for checking something like this:

```
let booted_incompatible = host
            .status
            .booted
            .as_ref()
            .map_or(false, |b| b.incompatible);
let staged_incompatible = host
                .status
                .staged
                .as_ref()
                .map_or(false, |b| b.incompatible);
if booted_incompatible || staged_incompatible {
   return Err...
}
```
or so?

File: lib/src/cli.rs (hunk 1 above)
Comment (ckyrouac) [reply lib/src/cli.rs]: :+1: thanks for the tip

File: lib/src/cli.rs (hunk 2 above)
Comment (ckyrouac) [reply lib/src/cli.rs]: sounds good, updated

**Review by cgwalters — APPROVED** [review]
Thanks!

**Pull request merged**